adapter/internal/oasparser/envoyconf/routes_with_clusters.go

```diff
@@ -996,108 +940,167 @@ end`
-			substitutionString = endpointBasepath + params.rewritePath
-		}
-	} else {
-		if strings.Contains(resourcePath, "?") {
-			resourcePath = strings.Split(resourcePath, "?")[0]
-		}
-		resourceRegex := generatePathRegexSegment(resourcePath)
-		substitutionString = generateSubstitutionString(resourcePath, endpointBasepath)
-		if strings.HasSuffix(resourcePath, "/*") {
-			resourceRegex = strings.TrimSuffix(resourceRegex, "((/(.*))*)")
-		}
-		pathRegex = "^" + basePath + resourceRegex
-	}
-	if xWso2Basepath != "" {
-		action = &routev3.Route_Route{
-			Route: &routev3.RouteAction{
-				HostRewriteSpecifier: hostRewriteSpecifier,
-				// TODO: (VirajSalaka) Provide prefix rewrite since it is simple
-				RegexRewrite: &envoy_type_matcherv3.RegexMatchAndSubstitute{
-					Pattern: &envoy_type_matcherv3.RegexMatcher{
-						EngineType: &envoy_type_matcherv3.RegexMatcher_GoogleRe2{
-							GoogleRe2: &envoy_type_matcherv3.RegexMatcher_GoogleRE2{
-								MaxProgramSize: nil,
-							},
-						},
-						Regex: pathRegex,
-					},
-					Substitution: substitutionString,
-				},
-				UpgradeConfigs:    getUpgradeConfig(apiType),
-				MaxStreamDuration: getMaxStreamDuration(apiType),
-				Timeout:           durationpb.New(time.Duration(adapterConfig.Envoy.Upstream.Timeouts.RouteTimeoutInSeconds) * time.Second),
-				IdleTimeout:       durationpb.New(time.Duration(adapterConfig.Envoy.Upstream.Timeouts.RouteIdleTimeoutInSeconds) * time.Second),
-			},
-		}
-	} else {
-		action = &routev3.Route_Route{
-			Route: &routev3.RouteAction{
-				HostRewriteSpecifier: hostRewriteSpecifier,
-			},
-		}
+	perRouteFilterConfigs := map[string]*any.Any{
+		wellknown.HTTPExternalAuthorization: extAuthzFilter,
+		wellknown.Lua:                       luaFilter,
 	}
 
-	headerBasedClusterSpecifier := &routev3.RouteAction_ClusterHeader{
-		ClusterHeader: clusterHeaderName,
-	}
-	action.Route.ClusterSpecifier = headerBasedClusterSpecifier
-	logger.LoggerOasparser.Debug("added header based cluster")
+	logger.LoggerOasparser.Debug("adding route ", resourcePath)
 
-	if (prodRouteConfig != nil && prodRouteConfig.RetryConfig != nil) ||
-		(sandRouteConfig != nil && sandRouteConfig.RetryConfig != nil) {
-		// Retry configs are always added via headers. This is to update the
-		// default retry back-off base interval, which cannot be updated via headers.
-		retryConfig := adapterConfig.Envoy.Upstream.Retry
-		commonRetryPolicy := &routev3.RetryPolicy{
-			RetryOn: retryPolicyRetriableStatusCodes,
-			NumRetries: &wrapperspb.UInt32Value{
-				Value: 0,
-				// If not set to 0, default value 1 will be
-				// applied to both prod and sandbox even if they are not set.
-			},
-			RetriableStatusCodes: retryConfig.StatusCodes,
-			RetryBackOff: &routev3.RetryPolicy_RetryBackOff{
-				BaseInterval: &durationpb.Duration{
-					Nanos: int32(retryConfig.BaseIntervalInMillis) * 1000,
-				},
-			},
-		}
-		action.Route.RetryPolicy = commonRetryPolicy
-	}
+	if resource.IsWithPolicies() {
```

Comment:
```suggestion
	if resource.ContainsPolicies() {
```

Reply:
I feel that the bool variable is stronger in the original way, where the resource don't just contain them rather policies are part of the resource (without them shouldn't work)

Reply:
What about hasPolicies?

Reply:
+1

Reply:
Fixed with https://github.com/wso2/product-microgateway/pull/2983/commits/a42b5aaed5ef8ade9850c3f3d0ea0e767661bef2